### PR TITLE
Add Touch ID Authentication Chapter

### PR DESCRIPTION
**Pull request by cocojoe**

### CODE REVIEW

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+---
+title: Touch ID Authentication
+description: This tutorial will show you how to utilise Touch ID and the Credentials Manager
+budicon: 345
+---
+
+<%= include('../../../_includes/_package', {
+  org: 'auth0-samples',
+  repo: 'auth0-ios-swift-sample',
+  path: '00-Credentials-TouchID',
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: should be -> 08-Credentials-TouchID

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+}) %>
+
+## Before Starting
+
+You should be familiar with previous tutorials. This tutorial assumes that you've integrated the [Auth0.swift](https://github.com/auth0/Auth0.swift/) dependencies in your project and you're familiar with presenting the Login. For further information, check out the [Login](/quickstart/native/ios-swift/00-login) quickstart.
+It is recommended that you are explore the [User Sessions](/quickstart/native/ios-swift/03-user-sessions) quickstart if you are not familiar with renewal of credentials using a `refreshToken`.
+
+## Touch ID Authentication
+
+Here's the scenario: You are using `webAuth` to present the HLP for the user to Login. After user authentication you want to store the user's credentials and use the `refreshToken` to renew the user's credentials without having to present the HLP. Additionally you want to utilize Touch ID to validate this renewal process.
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: Do you mention `HLP` somewhere before? If not, explain or use the large version `"Hosted Login Page" (HLP)`

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+You will to be using the [Credentials Manager](https://github.com/auth0/Auth0.swift/blob/master/Auth0/CredentialsManager.swift) utility in [Auth0.swift](https://github.com/auth0/Auth0.swift/) to streamline the management of user credentials and Touch ID.
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: remove `to`

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+### The Credentials Manager
+
+Add a property to your class for the credentials manager:
+
+```swift
+let credentialsManager: CredentialsManager!
+```
+
+Next, ensure the credentials manager is initialized in the appropriate `init` method for your class:
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: ~for~ _of_ your class

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    guard error == nil, let credentials = credentials else {
+        // Handle Error
+        // Route user back to Login Screen
+    }
+    // There is no need to store the credentials as you did in Login.  The Credentials Manager will do this for you internally
+    // Continue routing the user as authentication was a success
+}
+```
+
+If you were paying attention, you will have noticed there was no Touch ID prompt.
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: ~will~ _should_ have noticed

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+## Enable Touch ID
+
+The Credentials Manager can take care of this for you, once enabled. Go back to the snippet that initialized the Credentials Manager and add:
+
+```swift
+self.credentialsManager.enableTouchAuth(withTitle: "Touch to Authenticate")
+```
+
+Next time you call the `credentials` method, the user will be promoted for their Touch ID with the title "Touch to Authenticate".
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: typo ~promoted~ -> prompted

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+
+## Improving the User Experience
+
+What happens if the user has logged out and you have cleared the credentials from the Credentials Manager e.g.
+
+```swift
+self.credentialsManager.clear()
+```
+
+In this case the user will still be promoted for their touch and an error will be returned in the `credentials` closure as there
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: there's an extra new line?

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+are no credentials to renew from.
+
+The Credentials Manager has a `hasValid()` method that quickly lets you know if their are valid credentials that can be returned either directly or renewed and returned.
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: if ~their~ _there_ are??

**Review by lbalmaceda — CHANGES_REQUESTED** [review]

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md (hunk 7 above)
Comment (cocojoe) [reply articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: ?

Comment (lbalmaceda) [reply articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: Line 115 ends with "closure as there" 
Line 116 begins with "are no credentials to renew from."
I think you added a new line after the "there"

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+
+## Improving the User Experience
+
+What happens if the user has logged out and you have cleared the credentials from the Credentials Manager e.g.
+
+```swift
+self.credentialsManager.clear()
+```
+
+In this case the user will still be promoted for their touch and an error will be returned in the `credentials` closure as there
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: again ~promoted~ -> prompted

File: articles/quickstart/native/ios-swift/08-touch-id-authentication.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+## Improving the User Experience
+
+What happens if the user has logged out and you have cleared the credentials from the Credentials Manager e.g.
+
+```swift
+self.credentialsManager.clear()
+```
+
+In this case the user will still be promoted for their touch and an error will be returned in the `credentials` closure as there are no credentials to renew from.
```
Comment (lbalmaceda) [inline articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: this one is still present: 
~promoted~ -> prompted

Comment (cocojoe) [reply articles/quickstart/native/ios-swift/08-touch-id-authentication.md]: Good catch

**Review by lbalmaceda — APPROVED** [review]

**Comment by cocojoe** [comment]
@aaguiarz ping

**Pull request merged**